src/plugins/discover/public/application/view_components/canvas/index.tsx

```diff
@@ -41,6 +42,8 @@ export default function DiscoverCanvas({ setHeaderActionMenu, history, optionalR
       data,
     },
   } = useOpenSearchDashboards<DiscoverViewServices>();
+  const queryResultsBannerRef = useRef<HTMLDivElement>(null);
+  const extensionMap = data.query.queryString.getQueryResultService().getQueryResultExtensionMap();
```

Comment:
Can we add unit tests for this file? Not sure if @riysaxen-amzn was able to

Reply:
+1 it's kind of difficult to get the insight on how this one is being used. 

some unit tests and example implementation of this being used would help a lot.


apologies for the confusion but the original suggestion was to re-utilize the existing framework to achieve the same results. the existing framework for the query editor extensions could have been used to register a banner with functionality to accomplished what is required. the query editor is built with dataset selector. this is how we are able to avoid passing around stateful props.

but current implementation exposes a new service that now has it's own state and has to keep track of the state and communicate the state it's interacting with. which might cause issues anytime the UI state is out of sync or if a new feature is added.

the data plugin is more of a global plugin. Discover isn't the only plugin that uses it. Dashboards and visualizations in OSD core uses it as well. So plugins shouldn't register to a service that only ends up being renderable from the Discover page. So to futureproof it, for example if we wanted to toggle on visualizations or the workbench plugin to utilize your feature we wouldn't have to do new development again. It would just be just a opt in the top nav bar the data plugin search bar and we have access to all your features.

we should be able to take what you have and just register it towards the footer of the query editor. so to the user and to the mock up it looks the same but we are not creating a new service or new component with it's own state to manage.

please see this PR as well which is one way we can simplify same of the indexed view stuff https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8845